### DISCUSSION TITLE
Youtube downloader, to download from youtube music?

### DISCUSSION THREAD

**Discussion by IllMethods** [RandomNinjaAtk/docker-lidarr-extended#249] [Ideas]
Can this be done? I have YouTube premium, would like to be able to use YouTube music as a source as well. That would finish everyone's libraries.

Also, what if it searches track names for what it cant find as well as album title?

**Comment by RandomNinjaAtk** [comment]
#102

**Reply by RandomNinjaAtk** [reply]
If you can suggest a client, that can take a Tidal or Deezer  album URL and grab the files from youtube instead, I can look at incorporating it because it would be a simple to implement. If it cannot do that, I won't bother personally.

**Reply by RandomNinjaAtk** [reply]
This was implemented in the latest update. 

Set it to use deezer and don't input a arl token to use. Make sure require quality is disabled.

#102

**Reply by IllMethods** [reply]
Nice. Do I need the YouTube cookies.txt file for this to work? I can't seem to get this to work on unraid. I've tried putting the file in a lot of places without any luck.

**Reply by RandomNinjaAtk** [reply]
No, it does not require cookies.txt, that is only used for videos.

**Reply by IllMethods** [reply]
Mint. I guess I need to wait for my docker to be updated to see this.

**Reply by RandomNinjaAtk** [reply]
Yes, you have to be updated to the latest and greatest....

**Reply by IllMethods** [reply]
Ok, I do have it. I've just been watching the logs and every attempt is met with this. They all fail, | [✕] Failed to get album art [ENXIO] .


    ____                    
   / __/_______  __  _______
  / /_/ ___/ _ \/ / / / ___/
 / __/ /  /  __/ /_/ / /    
/_/ /_/   \___/\__, /_/     
              /____/ v0.9.1

freyr - (c) Miraculous Owonubi <omiraculous@gmail.com>
------------------------------------------------------
2023-06-18 23:16:06 :: Video :: 1.0.8 :: 349 of 745 :: Aaliyah :: Checking for IMVDB Slug
2023-06-18 23:16:06 :: Video :: 1.0.8 :: 349 of 745 :: Aaliyah :: IMVDB Slug :: aaliyah
2023-06-18 23:16:06 :: Video :: 1.0.8 :: 349 of 745 :: IMVDB :: Aaliyah :: Caching Records...
Checking directory permissions...[done] 
[deezer:album:106786562]
 [•] Identifying service...[Deezer]
 [•] Checking authentication...[authenticated]
 Detected [album]
 Obtaining album metadata...[done] 
  ➤ Album Name: Juicy
  ➤ Artist: Doja Cat
  ➤ Tracks: 1
  ➤ Type: Album
  ➤ Year: 2019
  ➤ Genres: R&B
 [•] Collating [Juicy]...
  [•] Inquiring tracks...[done] 
   • [01 Juicy]
      | ➤ Collating sources...
      |  ➤ [•] YouTube Music...[success, found 6 sources]
      | ➤ Awaiting audiofeeds...2023-06-18 23:16:11 :: Video :: 1.0.8 :: 349 of 745 :: IMVDB :: Aaliyah :: Chache is already up-to-date, skipping...
2023-06-18 23:16:11 :: Video :: 1.0.8 :: 349 of 745 :: Aaliyah :: Music Videos previously downloaded, skipping...
2023-06-18 23:16:11 :: Video :: 1.0.8 :: 350 of 745 :: Beyoncé :: Checking for IMVDB Slug
2023-06-18 23:16:11 :: Video :: 1.0.8 :: 350 of 745 :: Beyoncé :: IMVDB Slug :: beyonce
2023-06-18 23:16:11 :: Video :: 1.0.8 :: 350 of 745 :: IMVDB :: Beyoncé :: Caching Records...
[done] 
      | [✕] Failed to get album art [ENXIO] 
 [•] Download Complete
 [•] Embedding Metadata...
  • [✕] 01 Juicy [deezer:track:728314212] (failed: Error downloading album art [An error occurred while processing the data through the with middleware tagged [progressBar]: ENXIO: no such device or address, open '/dev/tty'])
[•] Collation Complete
============ Stats ============
 [•] Runtime: [6.8s]
 [•] Total queries: [01]
 [•] Total tracks: [01]
     » Skipped: [00]
     ✓ Passed:  [00]
     ✕ Failed:  [01]
 [•] Output directory: [/downloads-lidarr-extended/incomplete]
 [•] Total Output size: 0.00 B
 [•] Total Network Usage: 0.00 B
     ♫ Media: 0.00 B
     ➤ Album Art: 0.00 B
 [•] Output bitrate: 320k
===============================
2023-06-18 23:16:13 :: Audio :: 1.0.10 :: 2 :: missing :: 26 of 38 :: Doja Cat :: Juicy :: Single :: ERROR :: download failed, missing tracks...
2023-06-18 23:16:13 :: Audio :: 1.0.10 :: 2 :: missing :: 26 of 38 :: Doja Cat :: Juicy :: Single :: Retry Download in 1 second fix errors...
2023-06-18 23:16:15 :: Audio :: 1.0.10 :: 2 :: missing :: 26 of 38 :: Doja Cat :: Juicy :: Single :: Download Attempt number 2
[Info] ImportListSyncService: Starting Import List Sync

**Reply by RandomNinjaAtk** [reply]
Sometimes songs are not available. So you will have failures, which is completely normal based on my own experience. 

I'm not aware of anything that can be done about it.you'd have to troubleshoot more to investigate with the actual software client and take up any issues with the client's author.

**Reply by IllMethods** [reply]
It isn't the songs it can't find, just the album art which it then marks as failed and doesn't import

 | [✕] Failed to get album art [ENXIO] 
 [•] Download Complete
 [•] Embedding Metadata...
  • [✕] 01 MOOO! [deezer:track:547187822] (failed: Error downloading album art [An error occurred while processing the data through the with middleware tagged [progressBar]: ENXIO: no such device or address, open '/dev/tty'])

**Reply by RandomNinjaAtk** [reply]
Again, this is all handled by the client, nothing I can do about it... The script hands it off to the client, and the client takes care of everything... 

You can see that its all client side, by looking at the relevant code...
https://github.com/RandomNinjaAtk/docker-lidarr-extended/blob/64007230d31422a7891c7beb8565d60371b2934e/root/scripts/Audio.sh#L177-L179

Maybe the client has some additional options (flags/paramters) that could be used to help the situation, but I don't have time to investigate it further.

In my own personal testing, with 10 artists and a total of 347 possible albums. I was able to pull 178 albums successfully. Which I feel like is pretty good because I do not have a metric to compare it against the original and how much better it would have been. I suspect the actual source would work a bit better. But this is meant to be a fallback or a secondary option and its not going to be perfect...

**Reply by IllMethods** [reply]
Ok. I've just turned off pretag with beats. Maybe that is interfering with it. I'll let it run and let you know if that solved anything.

**Reply by RandomNinjaAtk** [reply]
Pretag has nothing to do with it because that happens after the download. And in your case, it sounds like the download is failing...
